ambry-replication/src/main/java/com.github.ambry.replication/ReplicaThread.java

```diff
@@ -115,6 +126,38 @@
     this.waitEnabled = !replicatingFromRemoteColo;
     this.replicatingOverSsl = replicatingOverSsl;
     this.datacenterName = datacenterName;
+    for (Map.Entry<DataNodeId, List<RemoteReplicaInfo>> entry : replicasToReplicateGroupedByNode.entrySet()) {
+      for (RemoteReplicaInfo info : entry.getValue()) {
+        replicatedPartitions.add(info.getReplicaId().getPartitionId());
+      }
+    }
+  }
+
+  /**
+   * Enables/disables replication on the given {@code id}.
+   * @param ids the {@link PartitionId}s to enable/disable it on.
+   * @param enable whether to enable ({@code true}) or disable.
+   */
+  void controlReplicationForPartitions(List<? extends PartitionId> ids, boolean enable) {
+    lock.lock();
+    try {
+      for (PartitionId id : ids) {
+        if (replicatedPartitions.contains(id)) {
+          if (enable) {
+            replicationDisabledPartitions.remove(id);
```

Comment:
Could you elaborate why the `if` is required? The doc of [`Set#remove`](https://docs.oracle.com/javase/7/docs/api/java/util/Set.html#remove(java.lang.Object)) says "`true` if this set contained the specified element" but this will surely be `true` because there is a preceding `contains()` check